packages/sitecore-jss/src/i18n/graphql.dictionary-service.ts

```diff
@@ -0,0 +1,225 @@
+import { GraphQLRequestClient } from '../graphql-request-client';
+import { SitecoreTemplateId } from '../constants';
+import { DictionaryPhrases, DictionaryServiceBase, CacheOptions } from './dictionary-service';
+
+// TODO: use graphql import instead of string (Anastasiya, March 2021)
+const query = `
+query DictionarySearch(
+    $rootItemId: String!,
+    $language: String!,
+    $dictionaryEntryTemplateId: String!,
+    $pageSize: Int = 10,
+    $after: String
+  ) {
+  search(
+    where: {
+      AND:[
+        { name: "_path",      value: $rootItemId },
+        { name: "_templates", value: $dictionaryEntryTemplateId },
+        { name: "_language",  value: $language }
+      ]
+    }
+    first: $pageSize
+    after: $after
+    orderBy: { name: "Title", direction: ASC }
+  ) {
+    total
+    pageInfo {
+      endCursor
+      hasNext
+    }
+    dictionaryPhrases: results {
+      key: field(name: "key") {
+        value
+      },
+      phrase: field(name: "phrase") {
+        value
+      }
+    }
+  }
+}
+`;
+
+export interface GraphQLDictionaryServiceConfig extends CacheOptions {
+  endpoint: string;
+  rootItemId?: string;
+  pageSize?: number;
+  appName: string;
```

Comment:
> For consistency with `GraphQLLayoutServiceConfig`, we should:
> 
> 1. rename `appName` > `siteName`
I agree that we should make things consistent. Given the this value comes from the property "appName" (defined in `package.json` of sample app), would it be more intuitive to rename siteName -> appName in layout service config instead?

> 2. add an option for overriding the query (i.e. `formatSearchQuery`)
See my response above about why query is not overridable.

Reply:
@anastasiya29 
1. I think it should be named according to the usage in the service. For both GraphQL and REST, layout and dictionary services, this parameter is fed into _site_ parameters. Plus, it'd be good to avoid another breaking change :)
2. 👍

Reply:
appName renamed to siteName. Related files (like the .json files for testing) have been updated too.